lib/src/header_parser/sub_parsers/structdecl_parser.dart

```diff
@@ -83,15 +91,30 @@ Struc? parseStructDeclaration(
     // Adding to seen here to stop recursion if a struct has itself as a
     // member, members are updated later.
     bindingsIndex.addStructToSeen(structUsr, _stack.top.struc!);
-    _setStructMembers(cursor);
   }
 
   if (bindingsIndex.isSeenStruct(structUsr)) {
-    _stack.top.struc = bindingsIndex.getSeenStruct(structUsr)!;
+    _stack.top.struc = bindingsIndex.getSeenStruct(structUsr);
+
+    final skipDependencies = _stack.top.struc!.parsedDependencies ||
+        (config.structDependencies == StructDependencies.opaque &&
+            pointerReference &&
+            ignoreFilter);
+
+    if (!skipDependencies) {
+      // Prevents infinite recursion if struct is member of itself.
```

Comment:
It can't be a member of itself (nesting). But a struct type can have a pointer member pointing to its own type.

Reply:
Yeah, definitely that's what I meant here. I should update the comment.